rancher/rancher
# Fix/issue 31336 (image list 500)

https://github.com/rancher/rancher/issues/31336

#  Problem
When users access `/v3/kontainerdrivers/rancher-images` in HA clusters the will receive a 500 if the request is served from any pod other than the master rancher pod.  This endpoint generates a list of images the user needs by crawling for images in the system catalog.  The system catalog is only available on the master pod.

# Solution
When we sync the system catalog we generate the image list and store it in a configuration map.  Then when any pod gets the request for the image list it can look it up in the configuration map. To do this i had to break out catalog/helm and catalog/git to their own package to avoid a cyclic dependency between the image and catalog package.

# Testing
1. install ha rancher on a new cluster
2. `GET /v3/kontainerdrivers/rancher-images` once for every rancher pod in the cluster
3. observe the request never fails.
